src/main/kotlin/com/terraformation/backend/customer/OrganizationService.kt

```diff
@@ -39,12 +40,34 @@ class OrganizationService(
     }
 
     return dslContext.transactionResult { _ ->
+      val isNewUser =
+          try {
+            userStore.fetchByEmail(email) == null
+          } catch (e: Exception) {
+            false
+          }
+
       val user = userStore.fetchOrCreateByEmail(email)
 
       organizationStore.addUser(organizationId, user.userId, role)
 
-      publisher.publishEvent(
-          UserAddedToOrganizationEvent(user.userId, organizationId, currentUser().userId))
+      if (isNewUser) {
```

Comment:
i debated adding this boolean into the UserAddedToOrganizationEvent but decided to create a new event for separation